navigator_helpers/tasks/sample_to_dataset/utils.py

```diff
@@ -0,0 +1,270 @@
+import io
+import json
+import re
+
+from pprint import pprint
+from typing import Tuple, Type, Union
+
+import pandas as pd
+
+from json_repair import repair_json
+from pydantic import BaseModel, ValidationError
+
+
+def extract_json(input_string: str) -> Union[dict, list]:
+    """
+    Extracts content between <json> and </json> tags from the input string.
+    If tags are not present, attempts to parse the entire input as JSON.
+
+    Args:
+        input_string (str): The input string containing JSON data, optionally within <json>...</json> tags.
+
+    Returns:
+        Union[dict, list]: The extracted or parsed JSON data, or an error message in a dict.
+    """
+
+    JSON_PATTERN = re.compile(r"<json>(.*?)</json>", re.DOTALL)
```

Comment:
I want to move some version of these extraction utilities into Jarvis ASAP! The tag approach is great.